src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,207 +1,850 @@
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[1], -t.hZ)) )), t)
+    poly_r = _transform_into_global_coordinate_system(Quadrangle{T}((
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], -t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(    t.r,  t.φ[2], +t.hZ)),
+        CartesianPoint(CylindricalPoint{T}(zero(T),  t.φ[2], +t.hZ)) )), t)
+    # normals of the surfaces show inside the volume primitives. 
+    (e_top, e_bot, mantle, poly_l, poly_r)
 end
 
+####################################################################
+####################################################################
+
+#=
+___
+   \
+    \
+_____\
+
+=#
+const VaryingCylinder{T,CO} = Cone{T,CO,Tuple{T,T},Nothing} # Full in φ
+const PartialVaryingCylinder{T,CO} = Cone{T,CO,Tuple{T,T},Tuple{T,T}}
+
+function _in(pt::CartesianPoint, c::VaryingCylinder{T,ClosedPrimitive}) where {T} 
+    az = abs(pt.z) 
+    (az <= c.hZ || csg_isapprox_lin(az, c.hZ)) && begin
+        r = hypot(pt.x, pt.y) 
+        rz = radius_at_z(c.hZ, c.r[1], c.r[2], pt.z)
+        r <= rz || csg_isapprox_lin(r, rz)
+    end
+end
+function _in(pt::CartesianPoint, c::VaryingCylinder{T,OpenPrimitive}) where {T} 
+    abs(pt.z) < c.hZ &&
+    hypot(pt.x, pt.y) < radius_at_z(c.hZ, c.r[1], c.r[2], pt.z)
+end
 
-#Constructors for Tubes
-Tube(;rMin = 0, rMax = 1, φMin = 0, φMax = 2π, zMin = -1/2, zMax = 1/2) = Cone(rMin, rMax, rMin, rMax, φMin, φMax, zMin, zMax)
-Tube(rMin, rMax, φMin, φMax, zMin, zMax) = Tube(; rMin = rMin, rMax = rMax, φMin = φMin, φMax = φMax, zMin = zMin, zMax = zMax)
+function surfaces(t::VaryingCylinder{T}) where {T}
+    bot_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), -t.hZ), t) 
+    top_center_pt = _transform_into_global_coordinate_system(CartesianPoint{T}(zero(T), zero(T), +t.hZ), t) 
+    mantle = ConeMantle{T,Tuple{T,T},Nothing,:inwards}((t.r[1], t.r[2]), t.φ, t.hZ, t.origin, t.rotation)
+    e_bot = CircularArea{T}(r = t.r[1], φ = t.φ, origin = bot_center_pt, rotation = t.rotation)
+    e_top = CircularArea{T}(r = t.r[2], φ = t.φ, origin = top_center_pt, rotation = -t.rotation * RotZ{T}(π))
+    # normals of the surfaces show inside the volume primitives. 
+    e_top, e_bot, mantle
+end
 
-function Tube(r::R, height::H) where {R<:Real, H<:Real}
-    T = float(promote_type(R,H))
-    Cone(T, T(r), nothing, T(height)/2)
+function _in(pt::CartesianPoint, c::PartialVaryingCylinder{T,ClosedPrimitive}) where {T} 
+    az = abs(pt.z) 
+    (az <= c.hZ || csg_isapprox_lin(az, c.hZ)) && begin
+        r = hypot(pt.x, pt.y) 
+        rz = radius_at_z(c.hZ, c.r[1], c.r[2], pt.z)
+        r <= rz || csg_isapprox_lin(r, rz)
+    end &&
+    _in_angular_interval_closed(atan(pt.y, pt.x), c.φ)
```

Comment:
Also here, `_in_angular_interval_closed` can just always return `true` if the angle is `nothing`.

```julia 
_in_angular_interval_closed(::T, ::Nothing) = true
```

This would reduce the number of lines of code here.

Reply:
So comment above.